src/components/Button/Button.tsx

```diff
@@ -179,6 +194,16 @@ class Button extends UIComponent<Extendable<IButtonProps>, any> {
       onClick(e, this.props)
     }
   }
+
+  private handleFocus = (e: React.SyntheticEvent) => {
+    const { onFocus } = this.props
+
+    this.setState({ isLastFocusFromKeyboard: whatInput.ask() === 'keyboard' })
```

Comment:
this line contains specific logic that will probably be reused in other components. Let's move it to a util function somewhere in the /lib directory, something like

```
  this.setState(focusFromKeyboardState())
```